src/EditorFeatures/Core.Wpf/InlineHints/InlineHintsTag.cs

```diff
@@ -253,5 +262,15 @@ private async Task StartToolTipServiceAsync(IToolTipPresenter toolTipPresenter)
 
             toolTipPresenter.StartOrUpdate(_textView.TextSnapshot.CreateTrackingSpan(_span.Start, _span.Length, SpanTrackingMode.EdgeInclusive), uiList);
         }
+
+        private void Adornment_MouseLeftButtonDown(object sender, MouseButtonEventArgs e)
+        {
+            if (e.ClickCount == 2)
+            {
+                e.Handled = true;
+                var replacementValue = _hint.ReplacementTextChange!.Value;
+                _ = _subjectBuffer.Replace(new VisualStudio.Text.Span(replacementValue.Span.Start, replacementValue.Span.Length), replacementValue.NewText);
```

Comment:
this span may be computed against a different snapshot version of hte buffer than the one currently displayed.  it woudl be probably hard to hit, but this could cause a crash.   You should validate that the span is still within the bounds of the buffer's current snapshot prior to making this change.